src/azul/indexer/document.py

```diff
@@ -289,18 +313,28 @@ class NullableString(FieldType[Optional[str], str]):
     null_string = '~null'
     es_type = 'string'
 
+    def __init__(self):
+        super().__init__(Optional[str], str)
+
     def to_index(self, value: Optional[str]) -> str:
         return self.null_string if value is None else value
 
     def from_index(self, value: str) -> Optional[str]:
         return None if value == self.null_string else value
 
+    @property
+    def api_type(self) -> JSON:
+        return make_type(str)
+
 
 null_str = NullableString()
 
 Number = Union[float, int]
 
-N_ = TypeVar('N_', bound=Number)
+# We do not specify a bound to this TypeVar due to PyCharm being unable to
+# infer return types for methods. If bound, PyCharm does not emit a warning
+# when passing a int to method of NullableNumber(float) or vise versa.
+N_ = TypeVar('N_')
```

Comment:
We should lead with an explanation of what this is, similar to N and T. Then describe the caveat.